### PR TITLE
Add some UI tests and reword error-code docs

### PR DESCRIPTION
**Pull request by Ezrashaw**
Added UI tests for `E0013` and `E0015`. Error code docs for `E0015` were a bit unclear (they referred to all non-const errors in const context, when only non-const functions applied), so I touched them up a bit.

I also fixed up some issues in the new `error_codes.rs` tidy check (linked #106341), that I overlooked previously.

r? @GuillaumeGomez

### CODE REVIEW

**Comment by Ezrashaw** [comment]
I think that I'll have some more of these coming so tell me what size PRs you want.

File: compiler/rustc_error_codes/src/error_codes/E0015.md [hunk 1]
```diff
@@ -1,33 +1,26 @@
-A constant item was initialized with something that is not a constant
-expression.
+A non-`const` function was called in a `const` context.
 
-Erroneous code example:
+Example of erroneous code:
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0015.md:3]: Based on the other code explanations (like [this one](https://github.com/rust-lang/rust/blob/master/compiler/rustc_error_codes/src/error_codes/E0005.md)), the actual correct sentence is "Erroneous code example").

File: src/tools/tidy/src/error_codes.rs [hunk 2]
```diff
@@ -174,22 +172,21 @@ fn check_error_codes_docs(
...
                 verbose,
                 "warning: Error code `{err_code}` uses the ignore header. This should not be used, add the error code to the \
                 `IGNORE_DOCTEST_CHECK` constant instead."
             );
         }
-        if emit_no_longer_warning {
+
+        if no_longer_emitted {
             no_longer_emitted_codes.push(err_code.to_owned());
-            verbose_print!(
```
Comment (GuillaumeGomez) [inline src/tools/tidy/src/error_codes.rs:188]: Nice catch!

**Comment by GuillaumeGomez** [comment]
> I think that I'll have some more of these coming so tell me what size PRs you want.

Like this is fine for me. Thanks for working on this!

File: compiler/rustc_error_codes/src/error_codes/E0015.md (hunk 1 above)
Comment (Ezrashaw) [reply compiler/rustc_error_codes/src/error_codes/E0015.md]: Done, thanks.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks! r=me once CI pass

**Pull request merged**